api/client/volume/remove.go

```diff
@@ -0,0 +1,44 @@
+package volume
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/client"
+	"github.com/docker/docker/cli"
+	"github.com/spf13/cobra"
+)
+
+func newRemoveCommand(dockerCli *client.DockerCli) *cobra.Command {
+	return &cobra.Command{
+		Use:     "rm VOLUME [VOLUME]...",
+		Aliases: []string{"remove"},
+		Short:   "Remove a volume",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := cli.MinRequiredArgs(args, 1, cmd); err != nil {
+				return err
+			}
+			return runRemove(dockerCli, args)
+		},
+	}
+}
+
+func runRemove(dockerCli *client.DockerCli, volumes []string) error {
+	client := dockerCli.Client()
+	var status = 0
+
+	for _, name := range volumes {
+		if err := client.VolumeRemove(context.Background(), name); err != nil {
```

Comment:
the context must be shared here. It's better to declare it before the `for` loop and pass it to `client.VolumeRemove`

```go
ctx := context.Background()

for _, name := range volumes{
    if err := client.VolumeRemove(ctx, …) //…
```